### PR TITLE
fix(platform): adding alt tags to the thumbnail component for accessibility

### PR DESCRIPTION
**Pull request by Lokanathan-k**
adding atl tags to the thumbnail component for accessibility

## Related Issue(s)
closes: https://github.com/SAP/fundamental-ngx/issues/6076

## Description

fixing the empty error message on required field and adding reset example to the form component.
##### During Implementation
1. Visual Testing:
- [x] visual misalignments/updates
- [x] check Light/Dark/HCB/HCW themes
- [x] RTL/LTR - proper rendering and labeling
- [x] responsiveness(resize)
- [x] Content Density (Cozy/Compact/(Condensed))
- [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [x] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [x] API boundary values
- [x] different combinations of components - free style
- [x] change the API values during testing
3. Documentation and Example validations
- [x] missing API documentation or it is not understandable
- [x] poor examples
- [x] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality
- [x] the commit message(s) follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

##### PR Review
- [x] visual misalignments/updates
- [x] check Light/Dark/HCB/HCW themes
- [x] RTL/LTR - proper rendering and labeling
- [x] responsiveness(resize)
- [x] Content Density (Cozy/Compact/(Condensed))
- [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [x] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [x] API boundary values
- [x] different combinations of components - free style
- [x] change the API values during testing
3. Documentation and Example validations
- [x] missing API documentation or it is not understandable
- [x] poor examples
- [x] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

### CODE REVIEW

File: libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
         [class.fdp-thumbnail-image--selected]="image.selected"
         [class.fdp-thumbnail-image--overlay]="image.overlayRequired"
         (click)="image.overlayRequired ? openDialog(image, mediaList) : thumbnailClick(image)"
```
Comment (dimamarksman) [inline libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.html:8]: you changed `thumbnailClick` signature, from now the first parameter must be event

File: libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
...
-        (keydown.space)="image.overlayRequired ? openDialog(image, mediaList) : thumbnailClick(image)"
+        (keydown)="image.overlayRequired ? openDialog(image, mediaList) : thumbnailClick($event, image)"
```
Comment (dimamarksman) [inline libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.html:9]: why did you remove `space`?
does it mean that any `keydown` will lead to `onClick` event firing?

**Review by Betrozov — APPROVED** [review]

File: libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.html [hunk 3]
```diff
@@ -5,9 +5,8 @@
         [class]="isHorizontal ? 'fdp-thumbnail-image--horizontal' : 'fdp-thumbnail-image'"
         [class.fdp-thumbnail-image--selected]="image.selected"
         [class.fdp-thumbnail-image--overlay]="image.overlayRequired"
-        (click)="image.overlayRequired ? openDialog(image, mediaList) : thumbnailClick(image)"
-        (keydown.space)="image.overlayRequired ? openDialog(image, mediaList) : thumbnailClick(image)"
-        tabindex="0"
+        (click)="image.overlayRequired ? openDialog(image, mediaList) : thumbnailClick($event, image)"
+        (keydown.space)="image.overlayRequired ? openDialog(image, mediaList) : thumbnailClick($event, image)"
```
Comment (dmitry-stepanenko) [inline libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.html:9]: maybe it would be better to have one method, that is used in both handlers since they're the same?

File: libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.ts [hunk 4]
```diff
@@ -95,7 +95,10 @@ export class ThumbnailImageComponent implements OnChanges, OnInit {
     }
 
     /** @hidden */
-    thumbnailClick(selectedMedia: Media): void {
+    thumbnailClick(event: KeyboardEvent, selectedMedia: Media): void {
```
Comment (dmitry-stepanenko) [inline libs/platform/src/lib/thumbnail/thumbnail-image/thumbnail-image.component.ts:98]: you are using this for `click` and `keydown` events, so event is either of type `KeyboardEvent` or `MouseEvent`

**Review by dmitry-stepanenko — COMMENTED** [review]
LGTM, just few minor comments

**Review by dmitry-stepanenko — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Comment by Sanjay-1312** [comment]
Issue 1: Fixed.

Issue 2: Fixed.

Issue 3: Not fixed.
image
OB: On tabbing to the last tab on further tab, the dialog opens automatically without user confirmation and once the dialog pop ups additional information is read.
EB: Without user confirmation the dialog shouldn't pop up and once the dialog pop ups only certain information should be read.
image
OB: On accessing any of the video read as 'Image' and on activating the Video the dialog using keyboard(Enter, Space bar), the dialog doesn't open and using the mouse the dialog pop ups.
EB: The UI type should be mentioned correctly to the user and the dialog should be activated using keyboard.

**Comment by Sanjay-1312** [comment]
Issue 3: 
image
Observed Behavior: On activating the image button, the dialog appears and additional information about dialog name is read multiple times.
Expected behavior: On appearing of the dialog only dialog name should be read once to the user.

Issue 5:
image
OB: On accessing any of the video read as 'Image' and on activating the Video the dialog using keyboard(Enter, Space bar), the dialog doesn't open and using the mouse the dialog pop ups.
EB: The UI type should be mentioned correctly to the user and the dialog should be activated using keyboard.

 Retest Details:
https://deploy-preview-7289--fundamental-ngx.netlify.app
JAWS 2021

**Comment by Lokanathan-k** [comment]
> Details

@Sanjay-1312 issue 3: when dialog content has focus by default the dialog header is read multiple time, we have discussed this issue when working on ATL issue for dialog component. this is a know issue.

**Comment by Sanjay-1312** [comment]
Hello Colleagues,

Issue 3: As per the above reply, on opening the dialog the dialog details is read multiple times and it is known issue.

**Comment by Karthickc1994** [comment]
Hello Colleague,

Retest has been taken the results are below:

Issue1: Fixed

Issue2:
Observed behaviour:
The list item within the groups are not accessible via arrow key. And the tutor message how to activate the list item information is not read out.
Expected behaviour:
The list item’s should be accessible via arrow keys. And the tutor message about how to access the list item should read out.

Screenshot:
image

Issue3:
Observed behaviour;
When accessing the video the information about the video is not read out.

Expected behaviour;
When accessing the video group information about the video should read out.
screenshot:
image

Retest Environment:
URL:https://fundamental-ngx.netlify.app/#/platform/thumbnail
Browser: Version 102.0.5005.63 (Official Build) (64-bit)
Screen Reader: JAWS2022.2204.20ILM

Regards,
Karthikeyan.c
